lib/Runtime/Base/ThreadContext.cpp

```diff
@@ -1277,9 +1277,10 @@ bool ThreadContext::IsActivePropertyId(Js::PropertyId pid)
 void ThreadContext::InvalidatePropertyRecord(const Js::PropertyRecord * propertyRecord)
 {
     InternalInvalidateProtoTypePropertyCaches(propertyRecord->GetPropertyId());     // use the internal version so we don't check for active property id
-
+#if ENABLE_NATIVE_CODEGEN
+    m_pendingJITProperties->Remove(propertyRecord);
```

Comment:
This was something I considered doing, but it seemed like a memory optimization (reclaiming in the runtime keeps the size bounded: on server we'll never hold more than the highest propertyid number of records), and I was a bit worried about increasing complexity/perf cost.

That said, upon reflection I think you're right. It's not much more effort, and it gives us more complete story for property records. I'll add the removal support in tomorrow morning.